server/src/main/java/org/elasticsearch/action/admin/indices/rollover/MetadataRolloverService.java

```diff
@@ -336,6 +336,15 @@ public Metadata.Builder withShardSizeForecastForWriteIndex(String dataStreamName
             shardCount += averageShardSize.numberOfShards();
         }
 
+        // Clear the previous forecast
+        if (dataStream.getIndices().size() > 1) {
+            final Index previousWriteIndex = dataStream.getIndices().get(dataStream.getIndices().size() - 2);
+            final IndexMetadata previousWriteIndexMetadata = metadata.getSafe(previousWriteIndex);
+            final IndexMetadata.Builder previousWriteIndexMetadataBuilder = IndexMetadata.builder(previousWriteIndexMetadata)
+                .shardSizeInBytesForecast(null);
```

Comment:
In the case where there is nothing in `ClusterInfo` for a non-write-index, would we then just fall back to the avg shard size recorded in the metadata? That seems OK to me, just wanted to be sure of the intention.

Reply:
I've reverted that part, I think that we should keep that information around? these shards are still taking space and we have to take that into account. cc @henningandersen

Reply:
Yeah, we may want to disregard this number though after rollover, since the shard size should then be more steady. But happy to keep it around and use it for now, it should not really make a big deal either way.